src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/VectorScoreScript.java

```diff
@@ -0,0 +1,149 @@
+package com.amazon.opendistroforelasticsearch.knn.plugin.script;
+
+import com.amazon.opendistroforelasticsearch.knn.index.util.KNNConstants;
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.script.ScoreScript;
+import org.elasticsearch.search.lookup.SearchLookup;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Map;
+
+/**
+ * Vector score script used for adjusting the score based on similarity space
+ * on a per document basis.
+ */
+public class VectorScoreScript extends ScoreScript {
+
+    private BinaryDocValues binaryDocValuesReader;
+    private final float[] inputVector;
+    private final String similaritySpace;
+
+    public float l2Squared(float[] queryVector, float[] inputVector) {
+        long squaredDistance = 0;
+        for (int i = 0; i < inputVector.length; i++) {
+            squaredDistance += Math.pow(queryVector[i]-inputVector[i], 2);
```

Comment:
Fixed. thanks

Reply:
Just out of curiosity, did you performance test the Math.fma computation as compared to the += diff * diff on an Intel CPU? This article implies that fma could be more performant as it can use a special intel instruction set: https://software.seek.intel.com/boosting-java-performance

Reply:
great ask!

I did test Math.fma and the performance appeared consistently slower than += on intel cpu with avx/avx2. It could be the overhead from underlying extra if, assert, double casting. More test results are always better.

Also note that on earlier cpu models, the fma implementation might be even worse. see https://stackoverflow.com/questions/44808081/what-are-the-accuracy-performance-benefits-of-using-math-fma